src/main/java/io/codelex/enums/practice/GameScissorsPaperStone.java

```diff
@@ -0,0 +1,93 @@
+package io.codelex.enums.practice;
+
+import java.util.Random;
+import java.util.Scanner;
+
+public class GameScissorsPaperStone {
+    static int countTrials = 0;
+    static int computerWon = 0;
+    static int youWon = 0;
+    static ScissorsPaperStone yourChoice;
+    static ScissorsPaperStone computerChoice;
+
+    public static void main(String[] args) {
+        System.out.println("Let us begin...");
+        beginGame();
+    }
+
+    private static void beginGame() {
+        System.out.println("Scissor-Paper-Stone");
+        yourTurn();
+    }
+
+    private static void yourTurn() {
+        System.out.print("\tYour turn (Enter s for scissor, p for paper, t for stone, q to quit): ");
+        Scanner in = new Scanner(System.in);
+        String choice = in.nextLine().toLowerCase();
+
+        switch (choice) {
+            case "q":
+                System.out.println();
+                System.out.printf("Number of trials: %d\n", countTrials);
+                System.out.printf("I won %d(%.2f%s). ", computerWon, (double)(computerWon * 100 / countTrials), "%");
+                System.out.printf("You won %d(%.2f%s).\n", youWon, (double)(youWon * 100 / countTrials), "%");
+                System.out.println("Bye!");
+                break;
+            case "s":
+                yourChoice = ScissorsPaperStone.SCISSORS;
+                computerTurn();
+                break;
+            case "p":
+                yourChoice = ScissorsPaperStone.PAPER;
+                computerTurn();
+                break;
+            case "t":
+                yourChoice = ScissorsPaperStone.STONE;
+                computerTurn();
+                break;
+            default:
+                System.out.println("\tInvalid input, try again...");
+                yourTurn();
+        }
+    }
+
+    private static void computerTurn() {
+        System.out.print("\tMy turn: " + randomChoice());
+        System.out.println();
+
+        resultOfWinner();
+        countTrials++;
+
+        System.out.println();
+        beginGame();
+    }
+
+    private static ScissorsPaperStone randomChoice() {
+        computerChoice = ScissorsPaperStone.class.getEnumConstants()[new Random().nextInt(ScissorsPaperStone.class.getEnumConstants().length)];
```

Comment:
var uzreiiz atgriezt, no need for assigment
var arī šo ielitk kā palīgmetodi enum klasē `getRandomChoice()` for example

Reply:
Paldies, izlaboju. Mēģināju visādi ar palīgmetodi enum klasē, lasot forumus, bet man pagaidām nesanāca, nolēmu paturēt esošo, kas strādā.

Reply:
https://stackoverflow.com/a/1972399
pēc šī noteikti sanāktu, bet noteikti ar nav tik svarīgi 😄

Reply:
Tieši no šī linka visus izprovēju un nesapratu, kāpēc nesanāk .